### PR TITLE
Drop MonadZero file, class, and instance

### LINKED ISSUE #67: Remove deprecated MonadZero instance (v0.15.0 PS)
When we make the next round of breaking changes (e.g. v0.15.0 of PureScript), this is a reminder to remove the MonadZero instance for Array. Relates to #66, which censors user-defined warnings due to the `MonadZero` instance. The build command should be updated to fail on any warning once we've removed the instance.

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Fixes #67 

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [x] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

**Comment by JordanMartinez** [comment]
:ping_pong: @thomashoneyman

This is where the `MonadZero` class is defined. Once this merges, all other libraries that have such instances will also need those instances removed to compile (e.g. `contrib` libraries).

**Comment by JordanMartinez** [comment]
So, should we wait until the end before merging this PR?

File: package.json [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
+    "build": "pulp build -- --censor-lib --strict",
     "test": "pulp test"
   },
   "devDependencies": {
     "pulp": "16.0.0-0",
     "purescript-psa": "^0.8.2",
     "rimraf": "^3.0.2"
   }
-}
+}
```
Comment (thomashoneyman) [inline package.json:13]:
```suggestion
}

```

**Comment by thomashoneyman** [comment]
No, I think this is fine to go ahead with. The contrib libraries won't depend on the new version of `control` until they're also being updated for 0.15, so it's not much of an issue.

**Review by thomashoneyman — APPROVED** [review]

**Pull request merged**